.github/workflows/pre-release.yaml

```diff
@@ -0,0 +1,64 @@
+# This workflow concerns the preparation of the `changeset` PR and keeping it updated by tracking the changes on `main` branch.
+name: Pre-Release Preparation
+
+on:
+  pull_request:
+    branches:
+      - main
+    types: [opened, reopened, labeled, synchronize]
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  pre-release:
+    name: Pre-Release Preparation
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
```

Comment:
combine checkout repo in the node setup action. you could rename it into a `setup-repo` action, and allow it to optionally setup `sui` as well